tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/generators/cpp/CppProtocolTestGenerator.java

```diff
@@ -0,0 +1,105 @@
+
+    protected final VelocityEngine velocityEngine;
+    private final ServiceModel serviceModel;
+    private final C2jTestSuite testModel;  // TODO: use an intermediate codegen model instead of raw C2J
+    private final String projectName;
+
+    public CppProtocolTestGenerator(ServiceModel serviceModel, C2jTestSuite testSuiteModel) throws Exception {
+        this.serviceModel = serviceModel;
+        this.testModel = testSuiteModel;
+        String prefix = testSuiteModel.getType().toString().toLowerCase();
+        this.projectName = testSuiteModel.getName() + "-" + prefix + "-protocol-tests";
+
+        velocityEngine = new VelocityEngine();
+        velocityEngine.setProperty(RuntimeConstants.RESOURCE_LOADERS, "classpath");
+        velocityEngine.setProperty("resource.loader.classpath.class", ClasspathResourceLoader.class.getName());
+        velocityEngine.addProperty(RuntimeConstants.RUNTIME_LOG_INSTANCE, new NOPLoggerFactory().getLogger(""));
+        velocityEngine.setProperty("context.scope_control.template", true);
+        velocityEngine.setProperty(RuntimeConstants.SPACE_GOBBLING, RuntimeConstants.SpaceGobbling.BC.toString());
+
+        velocityEngine.init();
+    }
+
+    public SdkFileEntry[] generateSourceFiles(ServiceModel dummy) throws Exception {
+        List<SdkFileEntry> fileList = new ArrayList<>();
+        fileList.add(generateCmakeFile());
+        fileList.add(generateTestDriver());
+
+        SdkFileEntry[] retArray = new SdkFileEntry[fileList.size()];
+        return fileList.toArray(retArray);
+    }
+
+    private SdkFileEntry generateCmakeFile() throws Exception {
+        VelocityContext context = createContext();
+        Template template = velocityEngine.getTemplate(CMAKE_LISTS_TEMPLATE, StandardCharsets.UTF_8.name());
+        return makeFile(template, context, "CMakeLists.txt", false);
+    }
+
+    protected SdkFileEntry generateTestDriver() throws Exception {
+        VelocityContext context = createContext();
+        Template template = velocityEngine.getTemplate(TEST_DRIVER_TEMPLATE, StandardCharsets.UTF_8.name());
+        return makeFile(template, context, "RunTests.cpp", true);
+    }
+
+    protected final VelocityContext createContext() {
+        VelocityContext context = new VelocityContext();
+        context.put("nl", System.lineSeparator());
+        context.put("projectName", projectName);
+        context.put("serviceModel", serviceModel);
+        context.put("testModel", testModel);
+        context.put("input.encoding", StandardCharsets.UTF_8.name());
+        context.put("output.encoding", StandardCharsets.UTF_8.name());
+        return context;
+    }
+
+    protected final SdkFileEntry makeFile(Template template, VelocityContext context, String path, boolean needsBOM) throws IOException {
+        StringWriter sw = new StringWriter();
+        template.merge(context, sw);
+
+        try {
+            sw.close();
```

Comment:
you should be using the [try with resources pattern](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) here i.e.

```
try(final StringWriter sw = new StringWriter();) {
    //...do stuff with sw
}

```